internal/iam/command_ipfilter_update.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/confluentinc/cli/v3/pkg/examples"
+)
+
+func (c *ipFilterCommand) newUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "update <id>",
+		Short:             "Update an IP filter.",
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validArgs),
+		RunE:              c.update,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Update the name and add an IP group to IP filter "ipf-abcde":`,
+				Code: `confluent iam ip-filter update ipf-abcde --name "New Filter Name" --add-ip-groups "ipg-12345"`,
+			},
+		),
+	}
+
+	cmd.Flags().String("name", "", "Updated name of the IP filter.")
+	pcmd.AddResourceGroupFlag(cmd)
+	cmd.Flags().StringSlice("add-ip-groups", []string{}, "Comma-separated list of IP groups to add.")
+	cmd.Flags().StringSlice("remove-ip-groups", []string{}, "Comma-separated list of IP groups to remove.")
+
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *ipFilterCommand) update(cmd *cobra.Command, args []string) error {
+	flags := []string{
+		"name",
+		"resource-group",
+		"add-ip-groups",
+		"remove-ip-groups",
+	}
+	if err := errors.CheckNoUpdate(cmd.Flags(), flags...); err != nil {
+		return err
+	}
+
+	filterName, err := cmd.Flags().GetString("name")
+	if err != nil {
+		return err
+	}
+
+	resourceGroup, err := cmd.Flags().GetString("resource-group")
+	if err != nil {
+		return err
+	}
+
+	addIpGroups, err := cmd.Flags().GetStringSlice("add-ip-groups")
+	if err != nil {
+		return err
+	}
+
+	removeIpGroups, err := cmd.Flags().GetStringSlice("remove-ip-groups")
+	if err != nil {
+		return err
+	}
+
+	currentIPFilterId := args[0]
+
+	// Get the current IP group we are going to be updating
+	currentIpFilter, err := c.V2Client.GetIamIpFilter(currentIPFilterId)
+	// Initialize our new IP groups list with the existing ids
+	var newIpGroupIds []string
+	for _, group := range currentIpFilter.GetIpGroups() {
+		newIpGroupIds = append(newIpGroupIds, group.GetId())
+	}
+
+	if err != nil {
+		return err
+	}
+
+	updateIpFilter := currentIpFilter
+
+	if filterName != "" {
+		updateIpFilter.FilterName = &filterName
+	}
+
+	if resourceGroup != "" {
+		updateIpFilter.ResourceGroup = &resourceGroup
+	}
+
+	// For each IP group ID being added that isn't in the existing slice, append it to the new slice
+	if len(addIpGroups) > 0 {
+		for _, ipGroupId := range addIpGroups {
+			if !slices.Contains(newIpGroupIds, ipGroupId) {
+				newIpGroupIds = append(newIpGroupIds, ipGroupId)
+			}
+		}
+	}
+	/*
+	 * For each IP group ID being removed that is in the existing slice, remove it from the slice.
+	 * This is accomplished by recreating the array with every element except for the one being removed
+	 */
+	if len(removeIpGroups) > 0 {
+		for _, ipGroupId := range removeIpGroups {
+			if slices.Contains(newIpGroupIds, ipGroupId) {
+				newIpGroupIds = removeIpFilterFromArray(newIpGroupIds, ipGroupId)
+			}
+		}
+	}
```

Comment:
I think this is a pattern we should expect to see in future commands, and it already exists for `confluent kafka quota update`. Can we make a generic function that can be used throughout the CLI, and unit test it?
```
func combineAddAndRemove(add, remove []string) []string
```

Reply:
(We have a `set` package that you might be interested in using for this, looks like your current solution is O(n^2) but that can be improved)